contracts/Bridge.sol

```diff
@@ -65,263 +27,117 @@ contract Bridge {
-        address[]             _noVotes;
-        DepositProposalStatus _status;
+        bytes32                  _dataHash;
+        address[]                _yesVotes;
+        address[]                _noVotes;
+        DepositProposalStatus    _status;
     }
 
     struct RelayerThresholdProposal {
-        uint                           _proposedValue;
+        uint256                        _proposedValue;
         mapping(address => bool)       _hasVoted;
         address[]                      _yesVotes;
         address[]                      _noVotes;
         RelayerThresholdProposalStatus _status;
     }
 
-    // chainID => number of deposits
-    mapping(uint => uint) public _depositCounts;
-    // chainID => depositNonce => GenericDepositRecord
-    mapping(uint => mapping(uint => GenericDepositRecord)) public _genericDepositRecords;
-    // chainID => depositNonce => ERC20DepositRecord
-    mapping(uint => mapping(uint => ERC20DepositRecord)) public _erc20DepositRecords;
-    // chainID => depositNonce => ERC721DepositRecord
-    mapping(uint => mapping(uint => ERC721DepositRecord)) public _erc721DepositRecords;
-    // chainID => depositNonce => CentrifugeAssetDepositRecord
-    mapping(uint => mapping(uint => CentrifugeAssetDepositRecord)) public _centrifugeDepositRecords;
-    // chainID => depositNonce => Proposal
-    mapping(uint => mapping(uint => DepositProposal)) public _depositProposals;
-    // destinationChainID => depositNonce => relayerAddress => bool
-    mapping(uint256 => mapping(uint256 => mapping(address => bool))) public _hasVotedOnDepositProposal;
-
-    event GenericDeposited(uint indexed destinationChainID, uint indexed depositNonce);
-    event ERC20Deposited(uint indexed destinationChainID, uint indexed depositNonce);
-    event ERC721Deposited(uint indexed destinationChainID, uint indexed depositNonce);
-    event CentrifugeAssetDeposited(uint indexed destinationChainID, uint indexed depositNonce);
-    event DepositProposalCreated(uint indexed destinationChainID, uint indexed depositNonce, bytes32 indexed dataHash);
-    event DepositProposalVote(uint indexed destinationChainID, uint indexed depositNonce, Vote indexed vote, DepositProposalStatus status);
-    event DepositProposalFinalized(uint indexed destinationChainID, uint indexed depositNonce);
+    // originChainHandlerAddress => number of deposits
+    mapping(address => uint256) public _depositCounts;
+    // originChainHandlerAddress => depositNonce => bytes
+    mapping(address => mapping(uint256 => bytes)) public _depositRecords;
+    // originChainID => originChainHandlerAddress => depositNonce => depositProposal
+    mapping(uint256 => mapping(address => mapping(uint256 => DepositProposal))) public _depositProposals;
+    // originChainID => originChainHandlerAddress => depositNonce => relayerAddress => bool
+    mapping(uint256 => mapping(address => mapping(uint256 => mapping(address => bool)))) public _hasVotedOnDepositProposal;
```

Comment:
Whats the logic of this?

Reply:
I'm assuming you're asking about the `_hasVotedOnDepositProposal` mapping?

This is an alternative to looping over the `_yesVotes` and `_noVotes` array of a proposal to check if an address has voted. With the assumption that we could have a great number of Relayers voting on proposals, I opted for a solution that would save a lot of gas if a contract ever needed to check if an address voted on a proposal